tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/docerina/FieldLevelDocsTest.java

```diff
@@ -0,0 +1,235 @@
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.docerina;
+
+import org.ballerinalang.docgen.docs.BallerinaDocGenerator;
+import org.ballerinalang.docgen.generator.model.DefaultableVarible;
+import org.ballerinalang.docgen.generator.model.Module;
+import org.ballerinalang.docgen.generator.model.Object;
+import org.ballerinalang.docgen.generator.model.Project;
+import org.ballerinalang.docgen.generator.model.Record;
+import org.ballerinalang.docgen.model.ModuleDoc;
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test class to check record/object field-level documentation in docerina.
+ */
+public class FieldLevelDocsTest {
+
+    private Module testModule;
+    // record with module-level documentation
+    private Record addressRecord;
+    // record with field-level documentation
+    private Record personRecord;
+    // record with both module-level & field-level documentation
+    private Record apartmentRecord;
+    // Object with module-level documentation
+    private Object studentObj;
+    // Object with field-level documentation
+    private Object teacherObj;
+
+    private static final String PARAGRAPH_OPEN_TAG = "<p>";
+    private static final String PARAGRAPH_CLOSE_TAG = "</p>";
+    private static final String NEW_LINE = "\n";
+    private static final String PARAGRAPH_CLOSE_WITH_NEW_LINE = PARAGRAPH_CLOSE_TAG + NEW_LINE;
+
+    @BeforeClass
+    public void setup() throws IOException {
+        String sourceRoot = "test-src/docerina/fieldLevelDocsProject";
+        CompileResult result = BCompileUtil.compile(sourceRoot, "testModule");
+
+        List<BLangPackage> modules = new LinkedList<>();
+        modules.add((BLangPackage) result.getAST());
+        Map<String, ModuleDoc> docsMap = BallerinaDocGenerator.generateModuleDocsFromBLangPackages(
+                Paths.get("src/test/resources", sourceRoot).toAbsolutePath().toString(), modules);
```

Comment:
@pramodya1994 is above path works on windows?

Reply:
Corrected